### PR TITLE
Modify S3649(multiple languages): Update to the education framework (APPSEC-184)

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**

### CODE REVIEW

File: rules/S3649/common/fix/secure-by-design.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+These provided APIs can be described as "secure by design". +
+By providing a builder pattern with parameter binding behind the scenes,
+`{secure_feature}` can be called "secure by design" as it adds
+multiple layers of security to the code while keeping the codebase shorter.
+
+*Note*: These types of APIs can also provide "raw" functions or equivalents. These
+functions allow developers to create complex queries using the user-friendly
+builder pattern. +
+These methods should be considered unsafe and should not be used with untrusted
+data.
```
Comment (gaetan-ferry-sonarsource) [inline rules/S3649/common/fix/secure-by-design.adoc:15]: Here, I would add a variable with an example of such an insecure function.
```suggestion
*Note*: These types of APIs can also provide "raw" functions or equivalents. These
functions allow developers to create complex queries using the user-friendly
builder pattern. +
These methods should be considered unsafe and should not be used with untrusted
data.
For example, `{secure_feature}` exposes `{unsafe_function}` that is prone to injections.
```

**Review by gaetan-ferry-sonarsource — APPROVED** [review]
LGTM with a minor suggestion.

File: rules/S3649/common/fix/secure-by-design.adoc (hunk 1 above)
Comment (loris-s-sonarsource) [reply rules/S3649/common/fix/secure-by-design.adoc:15]: Good idea!

File: rules/S3649/python/how-to-fix-it/sqlalchemy.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    users = metadata.tables['users']
+    conn = engine.connect()
+
+    sql = users.select().where(users.c.user == user)
+    conn.execute(sql)
+----
+
+=== How does this work?
+
+:secure_feature: SQLAlchemy
```
Comment (loris-s-sonarsource) [inline rules/S3649/python/how-to-fix-it/sqlalchemy.adoc:55]:
```suggestion
:secure_feature: SQLAlchemy
:unsafe_function:
```

**Pull request merged**